relayer/app/sender.go

```diff
@@ -15,19 +15,17 @@ import (
 	"github.com/omni-network/omni/lib/netconf"
 	"github.com/omni-network/omni/lib/tokens"
 	"github.com/omni-network/omni/lib/txmgr"
-	"github.com/omni-network/omni/lib/umath"
 	"github.com/omni-network/omni/lib/xchain"
 
 	"github.com/ethereum/go-ethereum"
 	"github.com/ethereum/go-ethereum/accounts/abi"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
 	ethtypes "github.com/ethereum/go-ethereum/core/types"
-	"github.com/ethereum/go-ethereum/params"
 )
 
 // Sender uses txmgr to send transactions a specific destination chain.
-type onSubmitFunc func(context.Context, *ethtypes.Transaction, *ethtypes.Receipt, xchain.Submission)
+type onSubmitFunc func(context.Context, ethclient.Client, *ethtypes.Transaction, *ethtypes.Receipt, xchain.Submission)
```

Comment:
why don't we replace the legacy `TransactionReceipt` with `OpTransactionReceipt` in `omni/lib/txmgr/txmgr.go:478`, then no need to query a second time?

Reply:
I think I'd prefer that. but blast radius of that change felt pretty big.  i'll give it a try